modules/imaging_uploader/test/imaging_uploaderTest.php

```diff
@@ -193,138 +144,41 @@ function testImagingUploaderFilterClearForm()
-
+        $this->_testPageUIs("/imaging_uploader/", $this->_loadingBrowseUI);
+        // click upload tab
+        $this->_testPageUIs("/imaging_uploader/#upload", $this->_loadingUploadUI);
     }
     /**
-     * Tests that, when uploading an empty file,
-     * some error text appears in the body.
-     *
-     * @return void
-     */
-    function testImagingUploaderwithoutData()
+      * This function could test UI elemnts in each Tabs.
+      *
+      * @param string $url this is for the url which needs to be tested.
+      * @param string $ui  UI elements in each Tabs.
+      *
+      * @return void
+      */
+    function _testPageUIs($url,$ui)
     {
-        $this->markTestSkipped("This method isn't working properly on travis.");
-        $this->safeGet($this->url . '/imaging_uploader/');
-        $this->webDriver->findElement(
-            WebDriverBy::ID("upload")
-        )->click();
-        $bodyText = $this->webDriver->findElement(
-            WebDriverBy::cssSelector("body")
-        )->getText();
-        $this->assertContains(
-            "mri_file: Make sure 'Are these Phantom Scans' is filled out.",
-            $bodyText
-        );
-    }
-    /**
-     * Tests that, when uploading a demo file,
-     * and the file name appears in the body.
-     *
-     * @return void
-     */
-    function testImagingUploaderwithData()
-    {
-        $this->markTestSkipped("This method isn't working properly on travis.");
-        $this->safeGet($this->url . '/imaging_uploader/');
-                $this->webDriver->findElement(
-                    WebDriverBy::Name("mri_file")
-                )->sendKeys("/TestVisitLabel.zip");
-                $this->webDriver->findElement(
-                    WebDriverBy::name("CandID")
-                )->sendKeys("999999");
-                $this->webDriver->findElement(
-                    WebDriverBy::name("PSCID")
-                )->sendKeys("8889");
-
-                $Visit_labelElement =  $this->safeFindElement(
-                    WebDriverBy::Name("Visit_label")
-                );
-                $Visit_label        = new WebDriverSelect($Visit_labelElement);
-                $Visit_label->selectByVisibleText("TestVisitLabel");
-
-                $PhantomElement =  $this->safeFindElement(
-                    WebDriverBy::Name("IsPhantom")
-                );
-                $Phantom        = new WebDriverSelect($PhantomElement);
-                $Phantom->selectByVisibleText("No");
-
-                $this->webDriver->findElement(
-                    WebDriverBy::ID("upload")
-                )->click();
-                $bodyText = $this->webDriver->findElement(
-                    WebDriverBy::cssSelector("body")
-                )->getText();
-                $this->assertContains("8889_999999_TestVisitLabel", $bodyText);
-
+        if ($url == "/imaging_uploader/") {
+            $this->safeGet($this->url . $url);
+            sleep(2);
```

Comment:
Please don't use sleep. Use exception catching instead

Reply:
safeGet() should be safe

Reply:
Otherwise, for the react table to load we could add a waitForElementToLoad($by) in the LorisIntegrationTest class.
Here is the example found at :  http://docs.seleniumhq.org/docs/04_webdriver_advanced.jsp#explicit-and-implicit-waits
```
WebDriver driver = new FirefoxDriver();
driver.get("http://somedomain/url_that_delays_loading");
WebElement myDynamicElement = (new WebDriverWait(driver, 10))
  .until(ExpectedConditions.presenceOfElementLocated(By.id("myDynamicElement")));
```